fedimint-core/src/db/mod.rs

```diff
@@ -1286,14 +1291,12 @@ impl<'tx> DatabaseTransaction<'tx> {
         's: 'a,
     {
         let decoders = self.decoders.clone();
-        let commit_tracker = self.commit_tracker.clone();
 
-        DatabaseTransactionRef(DatabaseTransaction {
+        DatabaseTransactionRef {
             tx: Box::new(&mut self.tx),
```

Comment:
Do we actually need this `Box<dyn IDatabaseTransaction + 'tx>` or could this be `&dyn IDatabaseTransaction + 'tx`?

Reply:
It would work in this case, but for other case we need to put a `PrefixDatabaseTransaction` which forces us to box it. &mut PrefixDatabaseTransaction` would cause a reference to a local object.

We could make it a `Cow` or something, but probably not worthwhile.